pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -171,30 +171,34 @@ func (f *Fit) Filter(ctx context.Context, cycleState *framework.CycleState, pod
 	return nil
 }
 
-func getErrReason(rn v1.ResourceName) string {
-	return fmt.Sprintf("Insufficient %v", rn)
-}
-
 // InsufficientResource describes what kind of resource limit is hit and caused the pod to not fit the node.
 type InsufficientResource struct {
 	ResourceName v1.ResourceName
-	Requested    int64
-	Used         int64
-	Capacity     int64
+	// We explicitly have a parameter for reason to avoid formatting a message on the fly
+	// for common resources, which is expensive for cluster autoscaler simulations.
+	reason    string
+	Requested int64
+	Used      int64
+	Capacity  int64
 }
 
 // Fits checks if node have enough resources to host the pod.
 func Fits(pod *v1.Pod, nodeInfo *schedulernodeinfo.NodeInfo, ignoredExtendedResources sets.String) []InsufficientResource {
-	return fitsRequest(computePodResourceRequest(pod), nodeInfo, ignoredExtendedResources)
+	r := fitsRequest(computePodResourceRequest(pod), nodeInfo, ignoredExtendedResources)
+	if len(r) == 0 {
+		return nil
+	}
+	return r
 }
 
 func fitsRequest(podRequest *preFilterState, nodeInfo *schedulernodeinfo.NodeInfo, ignoredExtendedResources sets.String) []InsufficientResource {
-	var insufficientResources []InsufficientResource
+	insufficientResources := make([]InsufficientResource, 0, 4)
 
 	allowedPodNumber := nodeInfo.AllowedPodNumber()
 	if len(nodeInfo.Pods())+1 > allowedPodNumber {
 		insufficientResources = append(insufficientResources, InsufficientResource{
 			v1.ResourcePods,
+			"Insufficient pods",
```

Comment:
Let's take the chance to fix this: `Too many pods`

Reply:
done.